spec/support/solidus_stripe/webhook/event_with_context_factory.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require "solidus_stripe/webhook/event"
+require "stripe"
+
+module SolidusStripe
+  module Webhook
+    # Factory to create a webhook event along with its context.
+    #
+    # It allows to create Stripe webhook from different sources (hash, stripe
+    # object) in different representations (json, stripe event object, header).
+    #
+    # The context for a event is composed by the timestamp and its secret, which
+    # in turn affect the header representation.
+    class EventWithContextFactory
+      def self.from_hash(hash:, timestamp: Time.zone.now, secret: "whsec_123")
+        new(hash: hash, timestamp: timestamp, secret: secret)
+      end
+
+      def self.from_object(object:, type:, timestamp: Time.zone.now, secret: "whsec_123")
+        hash_base = hash_base(object, type)
+        hash = hash_base.merge("webhook" => hash_base)
+        new(hash: hash, timestamp: timestamp, secret: secret)
+      end
+
+      def self.hash_base(object, type)
+        {
+          "id" => "evt_3MRUo1JvEPu9yc7w091rP2XV",
+          "object" => "event",
+          "api_version" => "2022-11-15",
+          "created" => 1_674_022_050,
+          "data" => {
+            "object" => object.as_json
+          },
+          "livemode" => false,
+          "pending_webhooks" => 0,
+          "request" => {
+            "id" => "req_3MRUo1JvEPu9yc7w0",
+            "idempotency_key" => "idempotency_key"
+          },
+          "type" => type
+        }
+      end
+      private_class_method :hash_base
+
+      attr_reader :hash, :timestamp, :secret
+
+      def initialize(hash:, timestamp: Time.zone.now, secret: "whsec_123")
+        @hash = hash
+        @timestamp = timestamp
+        @secret = secret
+      end
+
+      def object
```

Comment:
Similarly to comment about `hash` above, it could be more explicit to call this method and instance variable `event` or `stripe_event` rather than  `object` to make things more clear and specific

Reply:
Thanks for your suggestion, @loicginoux. The thing is that I want to encapsulate different representations of the Stripe event in different formats. So all of `json`, `hash` (now `data`), and `object` are stripe events. I used `object` as that's the Stripe naming for [its resources](https://www.rubydoc.info/gems/stripe/Stripe/StripeObject).

Maybe I could call it `instance`, but probably that would be very similar. Another option would be a verbose `stripe_object`. Thoughts?

Reply:
I'd ok with `stripe_object`, as that would probably remove the confusion you found. Waiting for your feedback.

Reply:
Thanks, @loicginoux. Done!